File: python/array_sort.py
""" Implements method sort  """
def sort(nums):
    """ recives an number array and returns the sorted array """
    for i in range(len(nums)):
        for j in range(i+1, len(nums)):
            if nums[i] > nums[j]:
                temp = nums[j]
                nums[j] = nums[i]
                nums[i] = temp

    return nums


if (__name__) == "__main__":
    print(sort([4,2,8,23,1,3,0,10,5,7,43,6,8,23,0,3,1,2,1,0]))  #[0,1,8,23,4,3,2,10,5,7,43,6,8,23]

File: c/array_sort.c
#include <stdio.h>

void sort(int *array, long int size) {

	for (int i = 0; i < size; i++) {
		int end = 1;
		for (int j =0; j < size-1; j++) {
			if (array[j] > array[j+1]) {
				int temp = array[j];
				array[j] = array[j+1];
				array[j+1] = temp;
				end = 0;
			}
		}
		if (end) break;
	}

}

int main(void) {
	int nums[] = {4, 6, 22, 11, 5, 7, 2, 1, 4, 3, 9, 7, 6, 8, 43, 12, 54, 34, 423, 543, 123, 32, 321, 75, 43, 6, 5, 1};

	long int n = sizeof(nums) / sizeof(nums[0]);

	sort(nums, n);

	for (int i = 0; i < n; i++) {
		printf("%d ", nums[i]);
	}

	printf("\n");

}
